app/lib/features/news/pages/add_news_page.dart

```diff
@@ -25,23 +28,52 @@ import 'package:flutter_svg/flutter_svg.dart';
 import 'package:go_router/go_router.dart';
 import 'package:mime/mime.dart';
 
+const addNewsKey = Key('add-news');
+
 class AddNewsPage extends ConsumerStatefulWidget {
-  const AddNewsPage({super.key});
+  const AddNewsPage({super.key = addNewsKey});
 
   @override
-  ConsumerState<ConsumerStatefulWidget> createState() => _AddNewsState();
+  ConsumerState<ConsumerStatefulWidget> createState() => AddNewsState();
 }
 
-class _AddNewsState extends ConsumerState<AddNewsPage> {
-  //General variable declaration
-  final textController = TextEditingController();
+class AddNewsState extends ConsumerState<AddNewsPage> {
+  EditorState textEditorState = EditorState.blank();
   NewsSlideItem? selectedNewsPost;
 
   //Build UI
   @override
   Widget build(BuildContext context) {
-    selectedNewsPost = ref.watch(newsStateProvider).currentNewsSlide;
+    ref.listen(newsStateProvider.select((i) => i.currentNewsSlide),
```

Comment:
ref.watch only leads to a re-render. But we need to re-establish the actual `EditorState`, so we instead we are listening to any change of the `currentNewsSlide` (but only that) and do the necessary changes